lib/nerves_hub_link/console_channel.ex

```diff
@@ -3,73 +3,63 @@ defmodule NervesHubLink.ConsoleChannel do
   `remote_iex` key to be enabled in the config:
   ```
   config :nerves_hub_link, remote_iex: true
   ```
 
-  Once connected, IO requests on the device will be pushed up the socket
+  Once connected, shell data on the device will be pushed up the socket
   for the following events:
 
-  * `get_line` - IO is requesting the next line from user input.
-  Typically just `iex () >`
-  * `put_chars` - Display the specified characters from the IEx Server for user review
-    * This requires an immediate reply of `:ok` and then IEx will send a `:get_line`
-    request to await the user input. NervesHubLinkWeb handles immediately replies `:ok`
-    to these events (see below)
-  * `init_attempt` - Pushed asynchronously after attempting to init an IEx Server.
-  Payload has a `success` key with a boolean value to specify whether the server
-  process was started successfully or not
-
-  The following events are supported _from_ the socket:
-
-  * `iex_terminate` - Kill any running IEx server in the channel
-  * `init` - The console channel starts without a linked IEx Server by default.
-  This must be called before sending I/O back and forth. Only one IEx Server is
-  initialized for this channel. If IEx Server has already been initialized and
-  is in a good state, then calling `init` will continue to use the linked session
-  and have no effect.
-  * `io_reply` - Send the reply characters to the IO. Requires specific keys in payload
-    * `kind` - event that you're replying to. Either `get_line` or `put_chars`
-    * `data` - characters to be put into the IO. `put_chars` requires this to be `ok` or `error`
+  The following events are supported _from_ the Server:
+
   * `phx_close` or `phx_error` - This will cause the channel to attempt rejoining
   every 5 seconds. You can change the rejoin timing in the config
   ```
   config :nerves_hub_link, rejoin_after: 3_000
   ```
+  * `restart` - Restart the IEx shell process
+  * `tty_text` - String data to send to the shell for evaluation
+  * `window_size` - A map with `:height` and `:width` keys for resizing the terminal
+
+  The following events are supported _from_ this client:
 
-  For more info, see [The Erlang I/O Protocol](http://erlang.org/doc/apps/stdlib/io_protocol.html)
+  * `tty_data` - String data to be displayed to the user console frontend
   """
 
   alias PhoenixClient.{Channel, Message}
   alias NervesHubLink.Client
 
   @rejoin_after Application.get_env(:nerves_hub_link, :rejoin_after, 5_000)
 
+  @version Mix.Project.config()[:version]
+
   defmodule State do
     defstruct socket: nil,
               topic: "console",
               channel: nil,
               params: [],
-              iex_pid: nil,
-              request: nil,
-              retry_count: 0
+              iex_pid: nil
   end
 
-  def start_link(opts), do: GenServer.start_link(__MODULE__, opts)
+  def start_link(opts), do: GenServer.start_link(__MODULE__, opts, name: __MODULE__)
```

Comment:
Not that it really matters, but what's causing this to become a singleton?

Reply:
O 🤦 

This is me doing `:sys.get_state(NervesHubLink.ConsoleChannel)` during troubleshooting/development. It doesn't need to be a singleton, its just convenient for my lazy self